lambdas/account-scoped/src/hrm/internalHrmRequest.ts

```diff
@@ -42,39 +43,87 @@ const requestFromInternalHrmEndpoint = async <TRequest, TResponse>(
     ...(body !== undefined && body !== null ? { body: JSON.stringify(body) } : {}),
   };
   console.debug(`Requesting ${endpointUrl} with options:`, options);
-  try {
-    const response = await fetch(endpointUrl, options);
-    if (!response.ok) {
+  let errorResult: ErrorResult<Error> = newErr<Error>({
+    message: 'Assert - error thrown without being set',
+    error: new Error(),
+  });
+  for (let attempt = 0; attempt < retryLimit; attempt++) {
+    try {
+      const response = await fetch(endpointUrl, options);
+      if (response.ok) {
+        const hrmResponse = await response.json();
+        console.info(`HRM responded ${response.status} to ${method} ${endpointUrl}`);
+        console.debug(`HRM response content:`, hrmResponse);
+        return newOk(hrmResponse as TResponse);
+      }
       const bodyText = await response.text();
-      return newErr<HttpClientError>({
+      errorResult = newErr<HttpClientError>({
         message: bodyText,
         error: new HttpClientError(bodyText, response.status),
       });
+    } catch (thrown) {
+      const error = thrown as Error;
+      errorResult = newErr<Error>({
+        message: error.message,
+        error,
+      });
+    }
+    console.warn(
+      `HRM request ${method} ${endpointUrl} attempt ${attempt + 1} failed`,
+      errorResult.message,
+      errorResult.error,
+    );
+    if (attempt + 1 < retryLimit) {
+      // eslint-disable-next-line @typescript-eslint/no-loop-func
+      await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
Maybe we want some back-off here? Not requiring a change just asking if it makes sense.
```suggestion
      await new Promise(resolve => setTimeout(resolve, 1000 * attempt));
```

Reply:
I just wanted to keep it simple and not delay the caller too long